e2e/examples/actions/redirect.spec.ts

```diff
@@ -0,0 +1,162 @@
+  await expect(firstNameInput).toHaveAttribute(
+    'aria-describedBy',
+    'errors-for-firstName',
+  )
+  await expect(emailErrors).toHaveText(
+    'String must contain at least 1 character(s)',
+  )
+  await expect(emailErrors).toHaveAttribute('role', 'alert')
+  await expect(emailInput).toHaveAttribute('aria-invalid', 'true')
+  await expect(emailInput).toHaveAttribute(
+    'aria-describedBy',
+    'errors-for-email',
+  )
+  await expect(firstNameInput).toBeFocused()
+
+  // Make first field be valid, focus goes to the second field
+  await firstNameInput.fill('John')
+  await submitButton.click()
+  await expect(firstNameInput).toHaveAttribute('aria-invalid', 'false')
+  await expect(emailInput).toBeFocused()
+
+  // Try another invalid message
+  await emailInput.fill('john')
+  await expect(page.locator('#errors-for-email')).toHaveText('Invalid email')
+  await expect(emailInput).toHaveAttribute('aria-invalid', 'true')
+
+  // Make form be valid
+  await emailInput.fill('john@doe.com')
+  await expect(emailInput).toHaveAttribute('aria-invalid', 'false')
+
+  // Submit form
+  await page.route('**/*', async (route) => {
+    await new Promise((f) => setTimeout(f, 100))
+    await route.continue()
+  })
+
+  submitButton.click()
+  await expect(submitButton).toBeDisabled()
+  await expect(page).toHaveURL('/success')
+})
+
+test('With JS disabled', async ({ browser }) => {
+  const context = await browser.newContext({ javaScriptEnabled: false })
+  const page = await context.newPage()
+
+  await page.goto('/examples/actions/redirect')
+
+  const firstNameInput = page.locator('input[name="firstName"]').first()
+  const emailInput = page.locator('input[name="email"]').first()
+  const submitButton = page.locator('form button:has-text("OK")').first()
+
+  // Server-side validation
+  await submitButton.click()
+
+  await page.reload()
+
+  const firstNameErrors = page.locator('#errors-for-firstName').first()
+
+  // Show field errors and focus on the first field
+  await expect(firstNameErrors).toHaveText(
+    'String must contain at least 1 character(s)',
+  )
+  await expect(firstNameErrors).toHaveAttribute('role', 'alert')
+  await expect(firstNameInput).toHaveAttribute('aria-invalid', 'true')
+  await expect(firstNameInput).toHaveAttribute(
+    'aria-describedBy',
+    'errors-for-firstName',
+  )
+
+  const emailErrors = page.locator('#errors-for-email').first()
+  const emailErrorDivs = await page.locator('#errors-for-email > div')
+  await expect(emailErrorDivs.first()).toHaveText(
+    'String must contain at least 1 character(s)',
+  )
+  await expect(emailErrorDivs.last()).toHaveText('Invalid email')
+  await expect(emailErrors).toHaveAttribute('role', 'alert')
+  await expect(emailInput).toHaveAttribute('aria-invalid', 'true')
+  await expect(emailInput).toHaveAttribute(
+    'aria-describedBy',
+    'errors-for-email',
+  )
+
+  await expect(await firstNameInput.getAttribute('autofocus')).not.toBeNull()
```

Comment:
```suggestion
  await expect(firstNameInput).toHaveAttribute('autofocus', 'true')
```

Reply:
This doesn't work. `autofocus` is either `null` or `''` 🤷